### PR TITLE
fix cross refs

### PR DESCRIPTION
**Pull request by jnavila**
fix transformation of crossrefs and add anchors to figures.

### CODE REVIEW

File: lib/tasks/book2.rake [hunk 1]
```diff
@@ -98,10 +98,10 @@ task :genbook2 => :environment do
           html.gsub!('<h1', '<h2')
           html.gsub!(/\/h1>/, '/h2>')
 
-          if xlink = html.scan(/\.html\#(.*?)\"/)
+          if xlink = html.scan(/href=\"1-.*?\.html\#(.*?)\"/)
```
Comment (peff) [inline lib/tasks/book2.rake]: This tightens up the regex, so I wondered if we would miss any other cases that did not point into the `1-` files. As it turns out there is only one such match, and it is a false positive that your change fixes! In A1.5 ("Git in Zsh"), the link to:

```
http://zsh.sourceforge.net/Doc/Release/User-Contributions.html#Version-Control-Information
```

is currently mangled into:

```
http://zsh.sourceforge.net/Doc/Release/User-Contributions/Version-Control-Information
```

which naturally gives a 404. With your patch, it's preserved correctly.

**Comment by peff** [comment]
I verified locally that this improves many of the reported cases (as well as the zsh link I mentioned). Thanks for tackling this!

**Comment by jnavila** [comment]
Thanks for merging! Could you run book generation for additional languages, please? At least, it would be great to have fr, ja, ko, nl and zh regenerated (they are fully translated).

**Comment by peff** [comment]
Sure, done for all those languages. I was a little hesitant to mass-build just because of the problems we ran into with `ja` recently. Rebuilding those 5, though, it seems to be the only one broken; I manually munged the database again to fix the problem.

**Comment by jnavila** [comment]
Thanks a lot! This will muffle the issues list.

**Comment by jnavila** [comment]
Thanks a lot. This should muffle a bit the issues.

**Pull request merged**